example/src/main/java/com/stripe/example/view/CreditCardView.java

```diff
@@ -0,0 +1,560 @@
+
+    private static int sScreenWidth = 0;
+
+    public static int getScreenWidth(Context context) {
+        if (sScreenWidth == 0) {
+            WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
+            DisplayMetrics dm = new DisplayMetrics();
+            wm.getDefaultDisplay().getMetrics(dm);
+            if (context.getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
+                sScreenWidth = Math.max(dm.widthPixels, dm.heightPixels);
+            } else {
+                sScreenWidth = Math.min(dm.widthPixels, dm.heightPixels);
+            }
+        }
+        return sScreenWidth;
+    }
+
+    public static int getImageResForCardBrand(String brand){
+        if (brand.equals(Card.AMERICAN_EXPRESS)) {
+            return R.drawable.stp_card_amex;
+        }
+        if (brand.equals(Card.DINERS_CLUB)) {
+            return R.drawable.stp_card_diners;
+        }
+        if (brand.equals(Card.DISCOVER)) {
+            return R.drawable.stp_card_discover;
+        }
+        if (brand.equals(Card.JCB)) {
+            return R.drawable.stp_card_jcb;
+        }
+        if (brand.equals(Card.MASTERCARD)) {
+            return R.drawable.stp_card_mastercard;
+        }
+        if (brand.equals(Card.VISA)) {
+            return R.drawable.stp_card_visa;
+        }
+        return R.drawable.stp_card_placeholder_template;
+
+    }
+
+    CustomHorizontalScrollView mScrollView;
+    Space mSpaceInContainer;
+    ImageView mIvCreditCardIcon;
+    EditText mEtNumber;
+    EditText mEtExpiryDate;
+    EditText mEtCvcNum;
+
+    private Callback mCallback;
+
+    private int mScrollViewWidth;
+    private int mScrollToPosition;
+    private boolean mCardNumberInView = true;
+    private int mColor;
+
+    private Card mCard;
+    private int mError;
+
+    public void setCallback(Callback callback) {
+        this.mCallback = callback;
+    }
+
+    public CreditCardView(Context context) {
+        super(context);
+        initView();
+    }
+
+    public CreditCardView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initView();
+    }
+
+    public CreditCardView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initView();
+    }
+
+    private void bindViews() {
+        mScrollView = (CustomHorizontalScrollView) findViewById(R.id.root_scroll_view);
+        mSpaceInContainer = (Space) findViewById(R.id.space_in_container);
+        mIvCreditCardIcon = (ImageView) findViewById(R.id.iv_credit_card_icon);
+        mEtNumber = (EditText) findViewById(R.id.et_credit_card_number);
+        mEtExpiryDate = (EditText) findViewById(R.id.et_expiry_date);
+        mEtCvcNum = (EditText) findViewById(R.id.et_cvc_num);
+    }
+
+    private void initView() {
+        inflate(getContext(), R.layout.view_credit_card, this);
+
+        // TODO : could also do this with ButterKnife
```

Comment:
We could indeed, but if this becomes a model for what to add to the library it would need to avoid that, as ButterKnife doesn't play well when the `R.id.x` aren't `final` (which it turns out they are not when you're compiling as a library)